### DISCUSSION TITLE
Acme not working with Custom DNS

### DISCUSSION THREAD

**Discussion by vikas85180** [polymind-inc/acmebot#710] [Q&A] [answered]
Hi, 

I have added the custom DNS provider configurations in the function app where the acmebot is hosted still I am not able to see my DNS zone in the list when I try to add a certificate.
- added all the required variable in the function app:
![image](https://github.com/shibayan/keyvault-acmebot/assets/57438692/77c365d8-b2a9-41e2-a078-42ebed818b61)

Still I can see in the browser console that it is picking up the default API URL for fetching the DNS zones:
![image](https://github.com/shibayan/keyvault-acmebot/assets/57438692/0e3f239d-431a-4b6c-a7af-e0471f2a3fad)

**Comment by shibayan** [comment]
Are the custom DNS endpoints implemented according to the Wiki specifications? They should be accessible as a REST API.

**Reply by vikas85180** [reply]
Hi Shibayan, thank you for the reply. Yes we have implemented the API endpoints as suggested in the WIKI, but after some research I observed that the endpoint for getting the DNS Zone as shown in the below SS is not similar to what is mentioned in you AcmeBot code. Please refer to the attached SS. Also yes we have implemented the endpoints as REST API but our DNS Zone in on premises, hence I have also added the VNet in the Function and I am able to resolve the DNS endpoint from the Function App where Dashboard is hosted.
![image](https://github.com/shibayan/keyvault-acmebot/assets/57438692/804a0e0d-3906-4d39-9a36-cd4fbd678013)
![image](https://github.com/shibayan/keyvault-acmebot/assets/57438692/02d2ef43-00b2-4890-a2c0-deea62e0e7ba)

I really appreciate you help!

**Reply by shibayan** [reply, marked as answer]
The Wiki specification is correct for implementing a custom DNS provider. Please refer to the following class implementation and sample code to check your implementation. Please refer to the Wiki for sample code, not for the implementation of Acmebot.

- https://github.com/shibayan/keyvault-acmebot/blob/master/KeyVault.Acmebot/Providers/CustomDnsProvider.cs
- https://github.com/shibayan/keyvault-acmebot/tree/master/samples/CustomDns
